### PR TITLE
Utility to unwrap a Decoder

### LINKED ISSUE #33: Utility to unwrap a Decoder function
In cases in which I wanted to implement a `FromJSON` instance by using this package combinators, I missed a utility to unpack the `Decoder` wrapped function:

```haskell
instance FromJSON SomeType where
  parseJSON = fromDecoder $ do
    ... -- Complicated decoding made easier with this package combinators
```

And implemented it myself, or wrote them in this style:

```haskell
instance FromJSON SomeType where
  parseJSON = f
    where
      Decoder f = do
        ... -- Complicated decoding made easier with this package combinators
```

Maybe I missed some already provided way to achieve this? Otherwise, I think that providing such a simple utility would be worth it.

### PR DESCRIPTION
**Pull request by aeqz**
Resolves #33

### CODE REVIEW

File: lib/Data/Aeson/Combinators/Decode.hs [hunk 1]
```diff
@@ -303,6 +304,17 @@ instance MonadFail Decoder where
...
 
+-- | Conversely, an Aeson's 'FromJSON' instance can be implemented by using 'Decoder' combinators.
+--
+-- > newtype People = People [Person]
+-- >
+-- > instance FromJSON People where
+-- >     parseJSON = fromDecoder $ Decode.list personDecoder
+fromDecoder :: Decoder a -> Value -> Parser a
+fromDecoder (Decoder f) = f
+{-# INLINE fromDecoder #-}
```
Comment (turboMaCk) [inline lib/Data/Aeson/Combinators/Decode.hs:315]: alternatively you can have also just change definition of type to:

```hs
newtype Decoder a =
Decoder { fromDecoder :: (Value -> Parser a) }
```

to get this function "for free". But there is no harm in having separate documentation from type for this so I'm OK with it.

**Review by turboMaCk — APPROVED** [review]
looks great

File: lib/Data/Aeson/Combinators/Decode.hs (hunk 1 above)
Comment (aeqz) [reply lib/Data/Aeson/Combinators/Decode.hs:315]: Yeah! In fact I wrote it this way to follow the same style as in the `Encode` module, which has a `run` function that unwraps an `Encoder`.

**Comment by turboMaCk** [comment]
released in 0.1.1.0

**Pull request merged**